### PR TITLE
Fix ComplexExpressionBuilder

### PR DESCRIPTION
**Pull request by ptomin**
- Create unit tests for `ComplexExpressionBuilder`
- Modify `ComplexExpressionBuilder`: return `<expression>` instead of `*&<expression>`

### CODE REVIEW

**Comment by ptomin** [comment]
`*&<exp> => <exp>` is a good improvement. Is not it, John?

File: subjects/Hunk-m68k/BENCHFN.c [hunk 1]
```diff
@@ -21,13 +21,13 @@ void fn00001000(int32 d0, int32 d7, byte * a0)
 		} while (d0_511 != 0x00);
 	}
 	*&globals->ptr3E70 = fp;
```
Comment (uxmal) [inline subjects/Hunk-m68k/BENCHFN.c]: Wonder why this didn't get fixed up?

File: subjects/Hunk-m68k/BYTEOPS.c [hunk 2]
```diff
@@ -21,13 +21,13 @@ void fn00001000(int32 d0, byte * a0)
 		} while (d0_508 != 0x00);
 	}
 	*&globals->ptr2B88 = fp;
```
Comment (uxmal) [inline subjects/Hunk-m68k/BYTEOPS.c]: Here too I expected your change to take effect

**Comment by uxmal** [comment]
Yeah, thanks for tracking that down. I did find some expressions in f3b483e that were not updated, strangely. Can you discover why?

**Comment by ptomin** [comment]
>  Can you discover why?

I'll investigate this tomorrow. 
Could you merge it? Any other changes could be done in other PR

**Comment by uxmal** [comment]
Sure thing. Nice work! You've done a great job sharpening the Reko type inference.

**Comment by ptomin** [comment]
> Yeah, thanks for tracking that down. I did find some expressions in f3b483e that were not updated, strangely. Can you discover why?

See #399

**Pull request merged**